File: apps/atalaya-web/src/app/core/models/telemetry-bucket.ts
/**
 * Punto agregado del histórico (downsampling, espejo de Atalaya.Contracts.TelemetryBucket).
 * Las métricas son el promedio del intervalo; se nombran igual que en TelemetryEvent para que el
 * gráfico y el selector de métrico se reusen sin cambios.
 */
export interface TelemetryBucket {
  ts: string;
  count: number;
  speedKmh: number;
  fuelPct: number;
  engineTempC: number;
}

File: libs/contracts/TelemetryBucket.cs
namespace Atalaya.Contracts;

/// <summary>
/// Punto agregado del histórico (<b>downsampling</b>, AUD-028): un intervalo de tiempo con el
/// <b>promedio</b> de cada métrica y el conteo de eventos del intervalo. Permite servir rangos largos
/// con un número acotado de puntos (≈ ancho del gráfico) en vez de miles de filas crudas
/// (ADR-005/007: el camino frío no compite con el caliente). Las métricas se nombran igual que en
/// <see cref="TelemetryEvent"/> para que el frontend reuse el mismo gráfico/selector.
/// </summary>
public sealed record TelemetryBucket(
    DateTimeOffset Ts,
    int Count,
    double SpeedKmh,
    double FuelPct,
    double EngineTempC);
